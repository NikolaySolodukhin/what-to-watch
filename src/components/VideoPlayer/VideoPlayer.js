import React from 'react';
import PropTypes from 'prop-types';

export default function VideoPlayer(props) {
  const {film, muted, controls} = props;
  return (
    <video src={film.video} poster={film.image} playsInline={true} width="280" height="175" muted={muted} controls={controls} autoPlay={true} />
  );
}

VideoPlayer.propTypes = {
  film: PropTypes.object.isRequired,
  muted: PropTypes.bool,
  controls: PropTypes.bool,
};
