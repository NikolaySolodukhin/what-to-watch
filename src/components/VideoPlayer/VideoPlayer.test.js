import React from 'react';
import renderer from 'react-test-renderer';

import VideoPlayer from './VideoPlayer';

describe(`<VideoPlayer/>`, () => {
  it(`VideoPlayer renders correctly`, () => {
    const film = {
      id: 1,
      title: `Fantastic Beasts`,
      image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
    };
    const tree = renderer
      .create(<VideoPlayer film={film} muted controls />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
