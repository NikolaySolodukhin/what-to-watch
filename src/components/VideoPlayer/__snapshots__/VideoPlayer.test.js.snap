// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlayer/> VideoPlayer renders correctly 1`] = `
<video
  autoPlay={true}
  controls={true}
  height="175"
  muted={true}
  playsInline={true}
  poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
  src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  width="280"
/>
`;
